src/commands/play.rs

```diff
@@ -50,45 +53,45 @@ pub async fn _play(
     // needed because interactions must be replied within 3s and queueing takes longer
     create_response(&ctx.http, interaction, SEARCHING).await?;
 
-    let enqueue_type = if url.contains("youtube.com/playlist?list=") {
-        EnqueueType::PLAYLIST
-    } else if url.starts_with("http") {
-        EnqueueType::URI
-    } else {
-        EnqueueType::SEARCH
+    let enqueue_type = match mode {
+        PlayMode::All | PlayMode::Reverse | PlayMode::Shuffle => EnqueueType::Playlist,
+        _ if url.contains("youtube.com/playlist?list=") => EnqueueType::Playlist,
+        _ if url.starts_with("http") => EnqueueType::Link,
+        _ => EnqueueType::Search,
     };
 
     let call = manager.get(guild_id).unwrap();
 
     match enqueue_type {
-        EnqueueType::URI => enqueue_song(&call, url.to_string(), true, flag).await,
-        EnqueueType::SEARCH => enqueue_song(&call, url.to_string(), false, flag).await,
-        EnqueueType::PLAYLIST => enqueue_playlist(&call, url).await,
+        EnqueueType::Link => enqueue_song(&call, url.to_string(), true, &mode).await,
+        EnqueueType::Search => enqueue_song(&call, url.to_string(), false, &mode).await,
+        EnqueueType::Playlist => enqueue_playlist(&call, url, &mode).await,
     };
 
     let handler = call.lock().await;
     let queue = handler.queue().current_queue();
     drop(handler);
 
     if queue.len() > 1 {
-        let estimated_time = calculate_time_until_play(&queue, flag).await.unwrap();
+        let estimated_time = calculate_time_until_play(&queue, &mode).await.unwrap();
```

Comment:
Make `calculate_time_until_play` take ownership of the mode. You will get "moved value" errors but you can derive the Copy trait for these enums since they are so "light".